lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -3521,6 +3530,28 @@ bool LLVMToSPIRVBase::isKnownIntrinsic(Intrinsic::ID Id) {
   }
 }
 
+// Add decoration if needed
+SPIRVInstruction *addFPBuiltinDecoration(SPIRVModule *BM, IntrinsicInst *II,
+                                         SPIRVInstruction *I) {
+  const bool AllowFPMaxError =
+      BM->isAllowedToUseExtension(ExtensionID::SPV_INTEL_fp_max_error);
+  bool IsLLVMFPBuiltin =
```

Comment:
llvm.fpbuiltin.* is a set of new LLVM builtins introduced in the open-source SYCL compiler (https://github.com/intel/llvm/pull/8134). It is not yet upstreamed to LLVM.org compiler. 
Support to translate these builtins currently rely on matching the name of the called function (as done here).
This matching will be replaced with matching with the actual Intrinsic ID if and when the front-end compiler change is available in LLVM.org.

@svenvh,

Can you please take a look at this and comment if this is an agreeable solution?

Thanks

Reply:
Perhaps it's worth starting an RFC thread on discourse.llvm.org first, to see if the intrinsics have a chance of getting accepted into llvm `main`.

Reply:
Hi @svenvh 

Thanks for comment. There is already a thread open here: https://discourse.llvm.org/t/rfc-floating-point-accuracy-control/66018. Discussion is ongoing.

Reply:
Thanks for pointing to the thread. I've skimmed over it, but it seems there hasn't been strong consensus yet and the discussion seems to have stalled. With that in mind, I'd probably prefer to avoid the `llvm.` prefix for the new "intrinsics", as they aren't LLVM intrinsics really.

Reply:
Hi @svenvh 

Thanks for looking at this. The builtin naming is actually coming from https://github.com/intel/llvm/pull/8134 and we expect this to get added to llorg at some point of time. The current PR does not have control over the naming.